src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs

```diff
@@ -208,6 +208,47 @@ public async Task GetAllOwnedCertificates_AllSliceStates(SliceState sliceState,
         certificates.Should().HaveCount(expectedCertificateCount);
     }
 
+
+    [Fact]
+    public async Task GetTop1ReceivedSlice_WhenNoSlices_ReturnNull()
+    {
+        await _connection.ExecuteAsync("DELETE FROM ReceivedSlices");
+
+        var slice = await _repository.GetTop1ReceivedSlice();
+
+        slice.Should().BeNull();
+    }
+
+    [Fact]
+    public async Task GetTop1ReceivedSlice_ReturnsSlice_AndRemoveSlice()
```

Comment:
Why did you add this test now?

Reply:
The method was untested after some previous changes.